src/otg/dataset/summary_statistics.py

```diff
@@ -0,0 +1,279 @@
+
+    @staticmethod
+    def _convert_odds_ratio_to_beta(
+        beta: Column, odds_ratio: Column, standard_error: Column
+    ) -> tuple:
+        """Harmonizes effect and standard error to beta.
+
+        Args:
+            beta (Column): Effect in beta
+            odds_ratio (Column): Effect in odds ratio
+            standard_error (Column): Standard error of the effect
+
+        Returns:
+            tuple: beta, standard error
+        """
+        # We keep standard error when effect is given in beta, otherwise drop.
+        standard_error = f.when(
+            standard_error.isNotNull() & beta.isNotNull(), standard_error
+        ).alias("standardError")
+
+        # Odds ratio is converted to beta:
+        beta = (
+            f.when(beta.isNotNull(), beta)
+            .when(odds_ratio.isNotNull(), f.log(odds_ratio))
+            .alias("beta")
+        )
+
+        return (beta, standard_error)
+
+    @staticmethod
+    def _calculate_confidence_interval(
+        pvalue_mantissa: Column,
+        pvalue_exponent: Column,
+        beta: Column,
+        standard_error: Column,
+    ) -> tuple:
+        """This function calculates the confidence interval for the effect based on the p-value and the effect size.
+
+        If the standard error already available, don't re-calculate from p-value.
+
+        Args:
+            pvalue_mantissa (Column): p-value mantissa (float)
+            pvalue_exponent (Column): p-value exponent (integer)
+            beta (Column): effect size in beta (float)
+            standard_error (Column): standard error.
+
+        Returns:
+            tuple: betaConfidenceIntervalLower (float), betaConfidenceIntervalUpper (float)
+        """
+        # Calculate p-value from mantissa and exponent:
+        pvalue = pvalue_mantissa * f.pow(10, pvalue_exponent)
+
+        # Fix p-value underflow:
+        pvalue = f.when(pvalue == 0, sys.float_info.min).otherwise(pvalue)
+
+        # Compute missing standard error:
+        standard_error = f.when(
+            standard_error.isNull(), f.abs(beta) / f.abs(pvalue_to_zscore(pvalue))
+        ).otherwise(standard_error)
+
+        # Calculate upper and lower confidence interval:
+        ci_lower = (beta - standard_error).alias("betaConfidenceIntervalLower")
+        ci_upper = (beta + standard_error).alias("betaConfidenceIntervalUpper")
+
+        return (ci_lower, ci_upper)
+
+    @classmethod
+    def from_ingested_gwas_summarystats(
+        cls: type[SummaryStatistics], session: Session, path: str
+    ) -> SummaryStatistics:
+        """Generate summary statistics dataset from ingested GWAS Catalog summary statistics.
+
+        Args:
+            session (Session): Session
+            path (str): path to ingested summary satistics in parquet format
+
+        Returns:
+            SummaryStatistics
+        """
+        # Raw sumstats schema:
+        gwas_sumstats_schema = t.StructType(
```

Comment:
For consistency, could you move this schema to a json? We can call it `raw_summary_stats.json` or something similar